docs/html/user_guide.rst

```diff
@@ -49,6 +49,22 @@ For more information and examples, see the :ref:`pip install` reference.
 .. _PyPI: https://pypi.org/
 
 
+Basic Authentication Credentials
+********************************
+
+pip supports basic authentication credentials. Basically, in the url there is
+a username and password separated by ``:``.
+
+  ::
+
+    https://[username[:password]]@pypi.company.com/simple
+    0123456789abcdef@pypi.company.com
+    https://aniruddha%24basak:gdg%24js%5Ejf%26l@pypi.company.com
```

Comment:
I think having this complicated example will make it harder to understand. I would have a separate paragraph talking specifically about escaping characters and then provide the example under it, like:

> Certain special characters are not valid in the authentication part of URLs. If the user or password part of your login credentials contain any of these special characters [here](/) then they must be percent-encoded. For example, for a user with username "user" and password "he//o" accessing a repository at pypi.company.com, the index URL with credentials would look like
>
>     https://user:he%2F%2Fo@pypi.company.com

Given the abundance of enterprise environments supporting older Pythons and pips, we may also want to provide some more details, like:

> Support for percent-encoded authentication in index URLs was added in pip 10.0.0 (in #3236). Users that **must** use authentication for their Python repository on systems with older pip versions should make the latest [get-pip.py](/) available in their environment to bootstrap pip to a recent-enough version.

Reply:
> Support for percent-encoded authentication in index URLs was added in pip 10.0.0 (in #3236). Users that must use authentication for their Python repository on systems with older pip versions should make the latest get-pip.py available in their environment to bootstrap pip to a recent-enough version.

@chrahunt I am having problem to understand this can you suggest me what should I do?

Reply:
I'm suggesting to include that text (or slightly re-worded if you see improvements) after the other text suggested above.